planner/core/logical_plan_builder.go

```diff
@@ -591,12 +591,10 @@ func (b *PlanBuilder) buildSelection(p LogicalPlan, where ast.ExprNode, AggMappe
 }
 
 // buildProjectionFieldNameFromColumns builds the field name, table name and database name when field expression is a column reference.
-func (b *PlanBuilder) buildProjectionFieldNameFromColumns(field *ast.SelectField, c *expression.Column) (colName, origColName, tblName, origTblName, dbName model.CIStr) {
-	if astCol, ok := getInnerFromParenthesesAndUnaryPlus(field.Expr).(*ast.ColumnNameExpr); ok {
-		origColName, tblName, dbName = astCol.Name.Name, astCol.Name.Table, astCol.Name.Schema
-	}
-	if field.AsName.L != "" {
-		colName = field.AsName
+func (b *PlanBuilder) buildProjectionFieldNameFromColumns(origField *ast.SelectField, colNameField *ast.ColumnNameExpr, c *expression.Column) (colName, origColName, tblName, origTblName, dbName model.CIStr) {
```

Comment:
Can we set `Schema` and `Table` for ColumnNameExpr when building `expression.Column` from it? Thus we can remove `c` here which will make the code clearer.

Reply:
I prefer not to change the ast's content :joy:

Reply:
It will not be changed.
Check https://github.com/pingcap/tidb/blob/master/planner/core/logical_plan_builder.go#L2119

Reply:
No, if there's no wild star. It's the original content.

Reply:
But we can create a new `ast.ColumnNameExpr` here? 
https://github.com/pingcap/tidb/blob/master/planner/core/logical_plan_builder.go#L1889

Reply:
I think we don't need to change it for now. The logic of creating the field names will be rewritten in the following PRs.